### DISCUSSION TITLE
Accessing Vehicle State Information for Controller Adjustment in Carla Simulator

### DISCUSSION THREAD

**Discussion by ouahid-ex9** [carla-simulator/carla#6834] [Q&A]
Hello Carla Simulator Community,

I am looking to control a vehicle using standard inputs like Steering, Gas, and Brake. For this, I require access to specific vehicle state data such as engine RPM, current gear ratio, wheel steering angle, gas pedal pressure, and brake pedal pressure. This information is essential to fine-tune the control commands based on the vehicle's current state. Could anyone guide me on how to access this data?

Thank you.

**Comment by awiswasi** [comment]
@ouahid-ex9 May I ask how you plan to use RPM and current gear ratio for example to adjust the control commands?
If you have ROS Bridge, you can get access to some of the data, such as max RPM, gear switch time, drag coefficient, etc.

**Reply by ouahid-ex9** [reply]
@awiswasi Actually, i need to have access to some informations about the vehicle state (rpm of the engine, current gear ratio, current steering angle, current brake torque, ...), because i want my custom controller needs those informations representing the current state of the vehicle, so i can give the appropriate commands to carla.VehicleControl method. For example, my controller needs the know the current engine rpm, in order to know the max torque the engine can develop (from torque-speed curve).

By the way, I think that unreal engine exposes the vehicle state ( engine, transmission, steering system, brakes, etc ) informations by default., correct me if i am wrong.
